Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -872,6 +872,9 @@ s32 WiiSockMan::AddSocket(s32 fd, bool is_rw)
     if (setsockopt(fd, SOL_SOCKET, SO_NOSIGPIPE, &opt_no_sigpipe, sizeof(opt_no_sigpipe)) < 0)
       ERROR_LOG_FMT(IOS_NET, "Failed to set SO_NOSIGPIPE on socket");
 #endif
+
+    int opt_broadcast = 1;
+    setsockopt(fd, SOL_SOCKET, SO_BROADCAST, (char*)&opt_broadcast, sizeof(opt_broadcast));
```

Comment:
I wrote the [hardware test](https://github.com/dolphin-emu/dolphin/files/10775111/wii-so-broadcast.zip) and Wii UDP sockets can use broadcast address by default. I tested a very simple scenario using a default socket rather than investigating the socket options _(which would have been way longer)_ by themselves.  The result on real hardware:
> ## HWTEST - UDP Broadcast
>
> Broadcast message sent successfully, ret=13
>
> Checking broadcast message to self:
> RET=13, MSG=Test message
> FAMILY=2, PORT=62623, ADDR=192.168.1.14

Here is a suggestion to address the side effect it can have on errno and also avoid altering TCP sockets.
```suggestion
    // Wii UDP sockets can use broadcast address by default
    if (!is_rw)
    {
      const auto state = Common::SaveNetworkErrorState();
      Common::ScopeGuard guard([&state] { Common::RestoreNetworkErrorState(state); });

      int socket_type;
      socklen_t option_length = sizeof(socket_type);
      const bool is_udp = getsockopt(fd, SOL_SOCKET, SO_TYPE, reinterpret_cast<char*>(&socket_type),
                                     &option_length) == 0 &&
                          socket_type == SOCK_DGRAM;
      const int opt_broadcast = 1;
      if (is_udp &&
          setsockopt(fd, SOL_SOCKET, SO_BROADCAST, reinterpret_cast<const char*>(&opt_broadcast),
                     sizeof(opt_broadcast)) != 0)
      {
        ERROR_LOG_FMT(IOS_NET, "Failed to set SO_BROADCAST on socket");
      }
    }
```

**NB:** This PR might not be enough to solve the broadcast issue in some circumstances. During my testing, I found out that Dolphin doesn't always use the right IP address on my machine to send broadcast packets _(using my VMware adapter instead of the Wi-Fi adapter)_. That part might have been addressed by https://github.com/dolphin-emu/dolphin/pull/9735.